src/main/java/org/javarosa/core/model/instance/ExternalDataInstance.java

```diff
@@ -23,17 +23,21 @@
  * @author ctsims
  */
 public class ExternalDataInstance extends DataInstance {
-    private String reference;
 
+    public static final String JR_SESSION_REFERENCE = "jr://instance/session";
+    public static final String JR_CASE_DB_REFERENCE = "jr://instance/casedb";
+    public static final String JR_LEDGER_DB_REFERENCE = "jr://instance/ledgerdb";
+    public static final String JR_SEARCH_INPUT_REFERENCE = "jr://instance/search_input";
```

Comment:
```suggestion
    public static final String JR_SEARCH_INPUT_REFERENCE = "jr://instance/search-input";
```
This got changed from a '-' to an '_' here: https://github.com/dimagi/commcare-core/commit/b1d80eed3e07dd8f5f2e039dcef493d6b43a9b26#diff-c023914dd76eddfd2ac9ce91bd463205b5258e85f05dffbc1826fa0548c32fd8R23